x-pack/legacy/plugins/siem/public/components/links/index.test.tsx

```diff
@@ -103,51 +117,89 @@ describe('Custom Links', () => {
 
   describe('ReputationLink', () => {
     test('it renders link text', () => {
-      const wrapper = mountWithIntl(
-        <ReputationLink domain={'192.0.2.0'}>{'Example Link'}</ReputationLink>
-      );
-      expect(wrapper.text()).toEqual('Example Link');
+      const wrapper = mountWithIntl(<ReputationLink domain={'192.0.2.0'} />);
+      expect(
+        wrapper
+          .find('EuiLink')
+          .at(0)
+          .childAt(0)
+          .text()
+      ).toEqual('virustotal.com, ');
+      expect(
+        wrapper
+          .find('EuiLink')
+          .at(1)
+          .childAt(0)
+          .text()
+      ).toEqual('talosIntelligence.com');
     });
 
     test('it renders correct href', () => {
-      const wrapper = mountWithIntl(
-        <ReputationLink domain={'192.0.2.0'}>{'Example Link'} </ReputationLink>
-      );
-      expect(wrapper.find('a').prop('href')).toEqual(
-        'https://www.talosintelligence.com/reputation_center/lookup?search=192.0.2.0'
-      );
-    });
+      const wrapper = mountWithIntl(<ReputationLink domain={'192.0.2.0'} />);
 
-    test("it encodes <script>alert('XSS')</script>", () => {
-      const wrapper = mountWithIntl(
-        <ReputationLink domain={"<script>alert('XSS')</script>"}>{'Example Link'}</ReputationLink>
-      );
-      expect(wrapper.find('a').prop('href')).toEqual(
-        "https://www.talosintelligence.com/reputation_center/lookup?search=%3Cscript%3Ealert('XSS')%3C%2Fscript%3E"
-      );
+      expect(
+        wrapper
+          .find('EuiLink')
+          .at(0)
+          .childAt(0)
+          .prop('href')
+      ).toEqual('https://www.virustotal.com/gui/search/192.0.2.0');
+      expect(
+        wrapper
+          .find('EuiLink')
+          .at(1)
+          .childAt(0)
+          .prop('href')
+      ).toEqual('https://talosintelligence.com/reputation_center/lookup?search=192.0.2.0');
     });
-  });
 
-  describe('VirusTotalLink', () => {
-    test('it renders sha passed in as value', () => {
-      const wrapper = mountWithIntl(<VirusTotalLink link={'abc'}>{'Example Link'}</VirusTotalLink>);
-      expect(wrapper.text()).toEqual('Example Link');
-    });
+    test("it encodes <script>alert('XSS')</script>", () => {
```

Comment:
consider adding an additional test to cover the case of data being injected via the new `{{ip}}` URL template syntax

Reply:
I'm adding some cases for invalid customised links